esmvalcore/preprocessor/_io.py

```diff
@@ -187,17 +188,110 @@ def load(file, callback=None, ignore_warnings=None):
     return raw_cubes
 
 
-def _by_two_concatenation(cubes):
-    """Perform a by-2 concatenation to avoid gaps."""
-    concatenated = iris.cube.CubeList(cubes).concatenate()
-    if len(concatenated) == 1:
-        return concatenated[0]
+def _concatenate_cubes(cubes, check_level):
+    """Concatenate cubes according to the check_level."""
+    kwargs = {
+        'check_aux_coords': True,
+        'check_cell_measures': True,
+        'check_ancils': True,
+        'check_derived_coords': True
+    }
```

Comment:
Wouldn't it be nicer to just allow specifying these keyword arguments in the recipe as arguments to the preprocessor function `concatenate` and then pass them on to `CubeList.concatenate`? That is conceptually simpler and will give users more control.

Reply:
Oh wait, that's no longer possible with the current design of `esmvalcore.dataset.Dataset.load`.